api/controller/auth.js

```diff
@@ -176,8 +176,34 @@ exports.login = function(req, res) {
  **/
 exports.signup = function(req, res) {
 
-	//insert the user using the same logic but without auth
-	users.addUser(req, res);
+	var user = JSON.parse(req.body.user);
+	if(user.beforeSignup) {
+		validateUsername(user.name, function(unique) {
+			if(unique) {
+				res.send(true);
```

Comment:
Send a more descriptive response instead of just `true`. Something like `{exists: false}`.

Reply:
Made the change in 1a7cc82.